internal/fs/vss_windows.go

```diff
@@ -845,14 +835,20 @@ func (p *VssSnapshot) Delete() error {
 
 		err = callAsyncFunctionAndWait(p.iVssBackupComponents.BackupComplete, "BackupComplete",
 			p.timeoutInMillis)
+		if err != nil {
+			return err
+		}
 
-		if _, _, err := p.iVssBackupComponents.DeleteSnapshots(p.snapshotID); err != nil {
-			err = newVssTextError(fmt.Sprintf("Failed to delete snapshot:  %s", err.Error()))
+		if _, _, e := p.iVssBackupComponents.DeleteSnapshots(p.snapshotID); e != nil {
+			err = newVssTextError(fmt.Sprintf("Failed to delete snapshot:  %s", e.Error()))
 			p.iVssBackupComponents.AbortBackup()
+			if err != nil {
```

Comment:
Shouldn't that return `e` not `err`? At that point `err` is always nil.

Reply:
No, because its wrapped using newVssTextError() to make the source of the error more explicit for the caller.

Reply:
Oh, sorry, you're right. I've just looked ad the highlighted changes...